### PR TITLE
Add thai keyboard distance

### PR DESCRIPTION
**Pull request by ppirch**
### What does this changes

Add function for calculate euclidean distance between two Thai characters on QWERTY Layout Keyboard

### CODE REVIEW

File: pythainlp/util/keyboard.py [hunk 1]
```diff
@@ -103,6 +103,20 @@
...
+  "ผ", "ป", "แ", "อ", "ิ", "ื", "ท", "ม", "ใ", "ฝ"
+]
+
+TH_SHIFT_QWERTY = [
+  "%", "+", "๑", "๒", "๓", "๔", "ู", "฿", "๕", "๖", "๗", "๘", "๙",
+  "๐", "\"", "ฎ", "ฑ", "ธ", "ํ", "๊", "ณ", "ฯ", "ญ", "ฐ", ",", "ฅ",
+  "ฤ", "ฆ", "ฏ", "โ", "ฌ", "็", "๋", "ษ", "ศ", "ซ", ".",
+  "(", ")", "ฉ", "ฮ", "ฺ", "์", "?", "ฒ", "ฬ", "ฦ"
+]
+
```
Comment (bact) [inline pythainlp/util/keyboard.py]: Do we suppose to have 4 rows for each keyboard layout here?

I'm not entirely sure about the current logic, but it seems to me these TH_QWERTY and TH_SHIFT_QWERTY have only one row (they're all 1-dimension array).

**Comment by bact** [comment]
Thanks for great contribution.
Do you mind to change the name of the layout from QWERTY to "Kedmanee" (or TIS 820 / TIS 820-2531 / TIS 820-2538, if that matter) ? 
https://en.wikipedia.org/wiki/Thai_Kedmanee_keyboard_layout

**Comment by ppirch** [comment]
> Thanks for great contribution.
> Do you mind to change the name of the layout from QWERTY to "Kedmanee" (or TIS 820 / TIS 820-2531 / TIS 820-2538, if that matter) ?
> https://en.wikipedia.org/wiki/Thai_Kedmanee_keyboard_layout

I agree to that suggestion. 
I changed the name from QWERTY to TIS 820-2538.

File: pythainlp/util/keyboard.py [hunk 2]
```diff
@@ -103,6 +103,20 @@
 EN_TH_TRANSLATE_TABLE = str.maketrans(EN_TH_KEYB_PAIRS)
 TH_EN_TRANSLATE_TABLE = str.maketrans(TH_EN_KEYB_PAIRS)
 
+TIS_820_2538 = [
+  "-", "ๅ", "/", "_", "ภ", "ถ", "ุ", "ึ", "ค", "ต", "จ", "ข", "ช",
```
Comment (wannaphong) [inline pythainlp/util/keyboard.py]: `"-", "ๅ", "/", "_"` -> `"_", "ๅ", "/", "-"`

Comment (ppirch) [reply pythainlp/util/keyboard.py]: It's fixed.

**Comment by wannaphong** [comment]
I have checked the algorithm. It's passed. `get_char_coord` output is x,y. The formula is formula={\sqrt%20{(x_{1}-x_{2})^{2}\plus(y_{1}-y_{2})^{2}}}.}&mode=inline)

**Review by bact — APPROVED** [review]
@ppirch I have committed my proposed code, please kindly check if I understand your algorithm correctly.

I basically changed the keyboard layout array from 1D to 2D.
Since the distance between Thai digit eight and nine should be different from the distance between Thai digit nine and zero.
These two pairs are both next to each other in 1D array, but it's not the case for 2D keyboard layout.

A new test case also added to illustrated this.

I also changed the constant names to refer to a modified TIS 820 2531 instead. Since that's more reflected the content inside the array.

File: pythainlp/util/keyboard.py (hunk 1 above)
Comment (ppirch) [reply pythainlp/util/keyboard.py]: I'm sorry it was my fault. It should be 2-dimension array. I forgot to put [] between each row.

**Review by wannaphong — APPROVED** [review]

**Comment by bact** [comment]
Sorry, I made a mistake in reverting the merge. Reverting the revert, and this PR is safely restored.

Thanks @ppirch again for the code.

**Pull request merged**